src/StackExchange.Exceptional.AspNetCore/ConfigSettings.ErrorStore.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.ComponentModel;
+using System.Configuration;
+using System.Runtime.Serialization;
+
+namespace StackExchange.Exceptional
+{
+    internal partial class ConfigSettings
+    {
+        /// <summary>
+        /// The ErrorStore section of the configuration, optional and will default to a <see cref="Stores.MemoryErrorStore"/> if not specified.
+        /// </summary>
+        public ErrorStoreSettings ErrorStore { get; set; } 
+
+        /// <summary>
+        /// A settings object describing an error store.
+        /// </summary>
+        public class ErrorStoreSettings
+        {
+            /// <summary>
+            /// The type of error store to use, File, SQL, Memory, etc.
+            /// </summary>
+            [JsonProperty(Required = Required.Always)]
+            public string Type { get; set; }
+
+            /// <summary>
+            /// The path to use on file based error stores.
+            /// </summary>
+            public string Path { get; set; }
+
+            /// <summary>
+            /// The connection string to use on database based error stores.  If provided, ConnectionStringName is ignored.
+            /// </summary>
+            public string ConnectionString { get; set; }
+
+            /// <summary>
+            /// The name of the connection string to use from the application's configuration.
+            /// </summary>
+            public string ConnectionStringName { get; set; }
+
+            /// <summary>
+            /// The size of this error log, either how many to keep or how many to display depending on type.
+            /// </summary>
+            public int Size { get; set; } = 200;
+
+            /// <summary>
+            /// The duration in seconds of error groups to roll-up, similar errors within this timespan will be shown as duplicates.
+            /// </summary>
+            public int RollupSeconds { get; set; } = 600;
```

Comment:
Let's make all of these nullable (`int?`), since they're just overrides to defaults here...we don't want to maintain defaults here as well.